packages/smart-account/src/SmartAccount.ts

```diff
@@ -561,13 +561,21 @@ class SmartAccount extends EventEmitter {
 
     if (updateImplTrx.data != '0x') {
       batchTrx.push(updateImplTrx, transactionDto.transaction)
-      response = this.sendTransactionBatch({ transactions: batchTrx })
+      response = this.sendTransactionBatch({
+        transactions: batchTrx,
+        paymasterServiceData: transactionDto.paymasterServiceData
+      })
     } else {
       // this case { if ( isUpdateImpTrx ) } will work only when user specifically wanted to just update Base wallet Implementation
       // if ( isUpdateImpTrx )
       // transactionDto.transaction = updateImplTrx
 
-      response = await aaSigner.sendTransaction(transactionDto.transaction, false, this)
+      response = await aaSigner.sendTransaction(
+        transactionDto.transaction,
+        false,
+        this,
+        transactionDto.paymasterServiceData
```

Comment:
more this as 2nd param. we can also make this an object and hence proper dto as there are more than 3 attributes

Reply:
Yup, moving to 2nd param for now, can be refactored in the redesign